### PR TITLE
feat(manager/helm-values): Add multi document file support 

### PR DESCRIPTION
**Pull request by robinschneider**
## Changes

Add support for multi-document files to the `helm-values` manager as discussed [here](https://github.com/renovatebot/renovate/discussions/24067)

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/helm-values/extract.spec.ts [hunk 1]
```diff
@@ -9,6 +9,10 @@ const helmMultiAndNestedImageValues = Fixtures.get(
   'multi_and_nested_image_values.yaml'
 );
 
+const multiDocumentFile = Fixtures.get(
```
Comment (viceice) [inline lib/modules/manager/helm-values/extract.spec.ts:12]: move to test case if only used once

File: lib/modules/manager/helm-values/extract.ts [hunk 2]
```diff
@@ -61,18 +61,20 @@ export function extractPackageFile(
   content: string,
   packageFile?: string
 ): PackageFileContent | null {
-  let parsedContent: Record<string, unknown> | HelmDockerImageDependency;
+  let parsedContent: Record<string, unknown>[] | HelmDockerImageDependency[];
   try {
     // a parser that allows extracting line numbers would be preferable, with
     // the current approach we need to match anything we find again during the update
-    // TODO: fix me (#9610)
```
Comment (viceice) [inline lib/modules/manager/helm-values/extract.ts:68]: revert

File: lib/modules/manager/helm-values/extract.ts [hunk 3]
```diff
@@ -61,18 +61,20 @@ export function extractPackageFile(
...
-    parsedContent = load(content, { json: true }) as any;
+    parsedContent = loadAll(content, null, { json: true }) as any;
   } catch (err) {
     logger.debug({ err, packageFile }, 'Failed to parse helm-values YAML');
     return null;
   }
   try {
-    const deps = findDependencies(parsedContent, []);
+    const deps: PackageDependency<Record<string, any>>[] = [];
+    parsedContent.forEach((content) => {
```
Comment (viceice) [inline lib/modules/manager/helm-values/extract.ts:75]: use for-of loop

**Comment by robinschneider** [comment]
@viceice Thank you for your feedback! I implemented it.

File: lib/modules/manager/helm-values/extract.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { load } from 'js-yaml';
+import {loadAll} from 'js-yaml';
```
Comment (viceice) [inline lib/modules/manager/helm-values/extract.ts:1]:
```suggestion
import { loadAll } from 'js-yaml';
```

File: lib/modules/manager/helm-values/__snapshots__/extract.spec.ts.snap
Comment (viceice) [inline lib/modules/manager/helm-values/__snapshots__/extract.spec.ts.snap:1]: revert this file

**Review by viceice — DISMISSED** [review]

**Comment by robinschneider** [comment]
@viceice Seems like the GitHub Actions hat some errors.

**Comment by viceice** [comment]
> @viceice Seems like the GitHub Actions hat some errors.

please don't merge main, that will block actions from first time contributions

**Comment by robinschneider** [comment]
> please don't merge main, that will block actions from first time contributions

Can you please explain what I should do different.

**Review by viceice — APPROVED** [review]

**Pull request merged**